docs/solvers/nn.rst

```diff
@@ -10,12 +10,25 @@ Neural Dual
 .. autosummary::
     :toctree: _autosummary
 
-    neuraldual.NeuralDualSolver
+    neuraldual.W2NeuralDual
+    neuraldual.plot_ot_map
+    neuraldual.plot_potential
 
-ICNN
-----
+Potentials
+----------
 .. autosummary::
     :toctree: _autosummary
 
-    icnn.ICNN
+    potential_base.PotentialBase
```

Comment:
I think this module is a bit too fragmented, would try to consolidate the classes into fewer files, e.g.:
```rst
.. autosummary::
    :toctree: _autosummary

    potentials.PotentialBase  # not sure if necessay
    potentials.ICNN
    potentials.MLP
    layers.PositiveDense
    layers.PosDefPotentials  # we might need to rename it + improve docstring
```
Am open to suggestions, also pinging @marcocuturi

Reply:
I'm open to putting the base, ICNN, MLP in the same class. Also regardless of that, is it somewhat confusing to have modules/objects approximately called potentials in `ott.problems.linear` and `ott.solvers.nn`?

Reply:
I agree with Michal on consolidation, and I think bundling MLP / ICNN and base makes sense indeed Brandon!

I do find it a bit confusing to have 2 references to potentials.

What I also find a bit confusing is calling `potential` something that's just a gradient of something (that may not be a potential).

Maybe call this simply `NN` or `Base` or `NNBase` ? Then from context people figure out whether it's a potential or something that looks like a map.

Reply:
What do you think if I create `ott.solvers.nn.models` and then refactor `nn.potential_base.PotentialBase` into `nn.models.NeuralModelBase`, and also put `ICNN` and `MLP` in there as subclasses?

Reply:
Just consolidated in https://github.com/ott-jax/ott/pull/219/commits/d7a948c56a923d4144eb66948c77257e24c8a543 and will push the updated notebooks soon once they finish running. Leaving this open in case there are any remaining changes you all want for this